pandas/core/indexes/base.py

```diff
@@ -573,7 +573,13 @@ def __array_wrap__(self, result, context=None):
 
         attrs = self._get_attributes_dict()
         attrs = self._maybe_update_attributes(attrs)
-        return Index(result, **attrs)
+        from pandas.core.dtypes.generic import ABCDatetimeIndex
```

Comment:
Can you import this at the top of the file where the other ABC's are being imported?